pandas/core/common.py

```diff
@@ -1403,6 +1404,16 @@ def _fill_zeros(result, x, y, name, fill):
                 if np.isinf(fill):
                     np.putmask(result,(signs<0) & mask, -fill)
 
+                # Floor division must be treated specially since NumPy
+                # causes 0//0 to be 0, but we want it to be NaN. (PR 9308)
+                if "floordiv" in name:
+                    if not isinstance(x, np.ndarray) and x == 0:
+                        nan_mask = mask
+                        np.putmask(result, nan_mask, np.nan)
+                    elif isinstance(x, np.ndarray):
+                        nan_mask = (mask & (x.ravel() == 0))
```

Comment:
hmm, any way to integrate this into the above section? don't like to special things if at all possible....